### PR TITLE
Snor-23 CSV parser with saving into databse

### PR DESCRIPTION
**Pull request by Pituuss**
Snor-23 CSV parser with saving into databse

### CODE REVIEW

File: src/main/java/com/mycompany/myapp/service/CSV/CSVParser.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    private static Boolean verify(List<ParsingContainer> parsedObjs, List<String> poolIdInDatabase) {
+        Map<String, ParsingContainer> objects =
+            parsedObjs.toLinkedMap(parsingContainer -> Tuple.of(parsingContainer.getPoolId(), parsingContainer));
+        return poolIdInDatabase
+            .forAll(poolId -> !objects.containsKey(poolId));
+    }
+
+    public static List<Pool> read(InputStream file, List<String> poolIdInDatabase) {
```
Comment (sjanikowski) [inline src/main/java/com/mycompany/myapp/service/CSV/CSVParser.java]: Maybe let's put public method above private methods it is using?

File: src/main/java/com/mycompany/myapp/service/CSV/ParsingContainer.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+import java.util.LinkedList;
+
+public class ParsingContainer {
+    private String poolId;
+    private String displayName;
+    private Integer maximumCount;
+    private Boolean enabled;
+    private String description;
+
+    public ParsingContainer() {
```
Comment (sjanikowski) [inline src/main/java/com/mycompany/myapp/service/CSV/ParsingContainer.java]: Do we follow any convention with fields and constructors formatting? We could use the one from DTO

File: src/main/java/com/mycompany/myapp/service/CSV/ParsingContainer.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        this.enabled = enabled;
+    }
+
+    public void setDescription(String description) {
+        this.description = description;
+    }
+
+    //    ugly but CSV file format is that ugly
+    public String getPool_id() {
+        return poolId;
```
Comment (sjanikowski) [inline src/main/java/com/mycompany/myapp/service/CSV/ParsingContainer.java]: This getter, if necessary, can call another getter if they return the same field

File: src/main/java/com/mycompany/myapp/service/CSV/PoolController.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import io.vavr.collection.List;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.IOException;
+
+@RestController
+public class PoolController {
```
Comment (sjanikowski) [inline src/main/java/com/mycompany/myapp/service/CSV/PoolController.java]: Isn't it out of scope of this task?

File: src/main/java/com/mycompany/myapp/service/CSV/CSVParser.java (hunk 1 above)
Comment (Pituuss) [reply src/main/java/com/mycompany/myapp/service/CSV/CSVParser.java]: Good point

File: src/main/java/com/mycompany/myapp/service/CSV/ParsingContainer.java (hunk 3 above)
Comment (Pituuss) [reply src/main/java/com/mycompany/myapp/service/CSV/ParsingContainer.java]: It isn't necessary, I must have created it on the run.

File: src/main/java/com/mycompany/myapp/service/CSV/ParsingContainer.java (hunk 2 above)
Comment (Pituuss) [reply src/main/java/com/mycompany/myapp/service/CSV/ParsingContainer.java]: Yup, this constructor is excessive.

**Pull request merged**